macros/edr/materializations/tests/test.sql

```diff
@@ -0,0 +1,242 @@
+  {% else %}
+    {% set test_result_metrics_rows = {} %}
+    {% for anomaly_score_row in anomaly_scores_rows %}
+      {% do test_result_metrics_rows.setdefault(anomaly_score_row.metric_name, []) %}
+      {% do test_result_metrics_rows[anomaly_score_row.metric_name].append(anomaly_score_row) %}
+    {% endfor %}
+
+    {% set elementary_test_results_rows = [] %}
+    {% for metric_rows in test_result_metrics_rows.values() %}
+      {% for metric_row in metric_rows %}
+        {% do elementary_test_results_rows.append(elementary.get_anomaly_test_result_row(flattened_test, metric_row, metric_rows)) %}
+      {% endfor %}
+    {% endfor %}
+    {% do elementary.cache_elementary_test_results_rows(elementary_test_results_rows) %}
+  {% endif %}
+
+  {% do context.update({"sql": elementary.get_anomaly_query(flattened_test)}) %}
+{% endmacro %}
+
+{% macro handle_schema_changes_test(flattened_test) %}
+  {% set schema_snapshots_tables_cache = elementary.get_cache("tables").get("schema_snapshots") %}
+  {% set schema_snapshots_table = elementary.get_elementary_test_table(flattened_test.name, 'schema_changes') %}
+  {% if schema_snapshots_table %}
+    {% do schema_snapshots_tables_cache.append(schema_snapshots_table) %}
+  {% endif %}
+
+  {% set elementary_test_results_rows = [] %}
+  {% set schema_changes_rows = elementary.query_test_result_rows() %}
+  {% for schema_changes_row in schema_changes_rows %}
+    {% do elementary_test_results_rows.append(elementary.get_schema_changes_test_result_row(flattened_test, schema_changes_row)) %}
+  {% endfor %}
+  {% do elementary.cache_elementary_test_results_rows(elementary_test_results_rows) %}
+{% endmacro %}
+
+{% macro handle_dbt_test(flattened_test) %}
+  {% set result_rows = elementary.query_test_result_rows(sample_limit=elementary.get_config_var('test_sample_row_count')) %}
+  {% set elementary_test_results_row = elementary.get_dbt_test_result_row(flattened_test, result_rows) %}
+  {% do elementary.cache_elementary_test_results_rows([elementary_test_results_row]) %}
+{% endmacro %}
+
+{% macro get_elementary_test_type(flattened_test) %}
+  {% if flattened_test.test_namespace == "elementary" %}
+    {% if flattened_test.short_name.endswith("anomalies") %}
+      {% do return("anomaly_detection") %}
+    {% elif flattened_test.short_name == 'schema_changes' %}
+      {% do return("schema_change") %}
+    {% endif %}
+  {% endif %}
+  {% do return("dbt_test") %}
+{% endmacro %}
+
+{% macro materialize_test() %}
+  {% set flattened_test = elementary.flatten_test(model) %}
+  {% set test_type = elementary.get_elementary_test_type(flattened_test) %}
+  {% set test_type_handler_map = {
+    "anomaly_detection": elementary.handle_anomaly_test,
+    "schema_change": elementary.handle_schema_changes_test
+  } %}
+  {% set test_type_handler = test_type_handler_map.get(test_type, elementary.handle_dbt_test) %}
```

Comment:
Thinking out loud:
To add a new type of test, we need to manually create a new one.
Don't we want to make it mandatory to add a new handler and extend the map?
Maybe we should throw an error if a new type is added and it has no reference in the type handler map (instead of trying to run a default handler)? That way we would make sure that we don't miss any step when adding a new test type, and it will help us understand what is missing as well while developing.
WDYT?